### PR TITLE
Add gpt-4o-mini as an available model

### LINKED ISSUE #14769: Add gpt-4o-mini model to assistant
### Check for existing issues

- [X] Completed

### Describe the feature

Add support for the new gpt-4o-mini model.  [API Docs here](https://platform.openai.com/docs/models/gpt-4o-mini) and [announcement with benchmarks here](https://openai.com/index/gpt-4o-mini-advancing-cost-efficient-intelligence/)

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by versecafe**
Release Notes:

- Added support for gpt-4o-mini ([#14769](https://github.com/zed-industries/zed/issues/14769), thanks [versecafe](https://github.com/versecafe)).

### CODE REVIEW

**Comment by versecafe** [comment]
Note I did not swap the default model to `gpt-4o-mini`, I think it should be but I'd rather get approval before making a change that would effect user defaults.

File: crates/open_ai/src/open_ai.rs [hunk 1]
```diff
@@ -100,6 +105,7 @@ impl Model {
             Model::Four => 8192,
             Model::FourTurbo => 128000,
             Model::FourOmni => 128000,
+            Self::FourOmniMini => 128000,
```
Comment (brianevanmiller) [inline crates/open_ai/src/open_ai.rs:108]: Self instead of `Model`?

Comment (versecafe) [reply crates/open_ai/src/open_ai.rs:108]: lmao, uh I mean, it works.... so yes? but I can go standardize that.

Comment (versecafe) [reply crates/open_ai/src/open_ai.rs:108]: Actually looking closer it should be using Self it's the ones using Model that don't fit in with the rest of the code

image

Comment (brianevanmiller) [reply crates/open_ai/src/open_ai.rs:108]: that's fair.  might as well refactor `serialize_model` line 119 to `Self` as well if you're already at it

Comment (versecafe) [reply crates/open_ai/src/open_ai.rs:108]: That's outside of the Model impl so needs the proper Model type, I could refactor it into model that would mean changing the structure of how it gets called across the project

Comment (brianevanmiller) [reply crates/open_ai/src/open_ai.rs:108]: I didn't notice that, got it

**Pull request merged**